### PR TITLE
DO NOT MERFGE: Attempt to fix memory 

### PR DESCRIPTION
**Pull request by younesbelkada**
Attempt to fix : https://github.com/huggingface/optimum-habana/issues/590

cc @regisss 
Do not merge for now

### CODE REVIEW

**Comment by younesbelkada** [comment]
cc @vivekgoe this might fix the bug you tagged me, can you try to have a look 🙏 ?

**Comment by vivekgoe** [comment]
@younesbelkada this PR does reduce memory usage, but with this loss is not converging.  

Can the issue be due to linear layer used in the model being a derived class from nn.linear? 
https://github.com/huggingface/transformers/blob/09eb11a1bd6e77833fde7020efc57e6a804e4e8a/src/transformers/models/falcon/modeling_falcon.py#L75

If I replace all instances of FalconLinear in the model with nn.Linear then memory footprint goes back to numbers before. Also this memory increase is not seen for other models using PEFT.
